pkg/system/phase4_configuring.go

```diff
@@ -541,16 +545,109 @@ func (r *Reconciler) awaitEndpointDeploymentPods() error {
 func (r *Reconciler) ReconcileHPAEndpoint() error {
 	// Wait for the the endpoint deployment to become ready
 	// only if HPA was not created yet
+	log := r.Logger.WithField("func", "ReconcileHPAEndpoint")
 	if r.HPAEndpoint.UID == "" {
 		if err := r.awaitEndpointDeploymentPods(); err != nil {
 			return err
 		}
 	}
+	var err error
+	prometheusList := deployedPrometheus(log)
+	switch r.NooBaa.Spec.AutoscalingType {
+	case "keda":
+		err = r.autoscaleKeda(prometheusList)
+	case "hpav2":
+		//TODO
+		log.Infof("HPAV2 implementation pending")
+	default:
```

Comment:
instead of `default` use a specific handling of `hpav1`. if `r.NooBaa.Spec.AutoscalingType` is not specified, use `options.DefaultAutoscalingType`